### PR TITLE
feat: DmnModelUtilTest add test

### PR DESCRIPTION
**Pull request by KarpovIvan**
## Description

task: [24657](https://github.com/camunda/camunda/issues/24657)

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+              </outputEntry>
+            </rule>
+          </decisionTable>
+        </decision>
+      </definitions>
+      """;
+
+  @Test
+  void parseDmnModelShouldReturnModelInstanceWhenValidXml() {
+    // Given: A valid DMN XML string
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:58]: nitpick, but can you do `given`, `when`, `then` (lowercase) for consistency with the rest of the repo. Same applies for all tests that you are writing

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                <text>"Jedi"</text>
+              </outputEntry>
+            </rule>
+          </decisionTable>
+        </decision>
+      </definitions>
+      """;
+
+  @Test
+  void parseDmnModelShouldReturnModelInstanceWhenValidXml() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:57]:
```suggestion
  void shouldParseValidXml() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    String dmnXml = "<invalid></invalid>";
+
+    // When & Then: parseDmnModel should throw OptimizeRuntimeException
+    assertThatThrownBy(() -> DmnModelUtil.parseDmnModel(dmnXml))
+        .isInstanceOf(DmnModelException.class)
+        .hasMessage("Unable to parse model");
+  }
+
+  @Test
+  void extractDecisionDefinitionNameShouldReturnNameWhenDecisionExists() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:79]:
```suggestion
  void shouldExtractDecisionNameWhenExists() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+    // When: extractDecisionDefinitionName is called
+    Optional<String> result = DmnModelUtil.extractDecisionDefinitionName(decisionKey, DMN_V1);
+
+    // Then: The name of the decision should be returned
+    assertThat(result).isPresent().get().isEqualTo("Jedi or Sith");
+  }
+
+  @Test
+  void extractDecisionDefinitionNameShouldReturnEmptyWhenDecisionNotFound() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:91]:
```suggestion
  void shouldReturnEmptyWhenDecisionKeyNotFound() {
```

Notice the format (`should...`). Please apply universally for all test names you write

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 1 above)
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:58]: I also don't think the `: ...` part is needed anywhere, as tests should mostly be self-documenting by their name, variables, and structure. Can you remove from all methods please?

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    // When: extractDecisionDefinitionName is called
+    Optional<String> result = DmnModelUtil.extractDecisionDefinitionName(decisionKey, dmnXml);
+
+    // Then: The result should be empty
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  void extractInputVariablesShouldThrowExceptionWhenDecisionKeyIsNull() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:107]:
```suggestion
  void shouldThrowExceptionWhenDecisionKeyIsNull() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    DmnModelInstance model = Mockito.mock(DmnModelInstance.class);
+
+    // When & Then: extractInputVariables should throw IllegalArgumentException
+    assertThatThrownBy(() -> DmnModelUtil.extractInputVariables(model, null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("decisionKey must not be null");
+  }
+
+  @Test
+  void extractOutputVariablesShouldThrowExceptionWhenDecisionKeyIsNull() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:118]:
```suggestion
  void shouldThrowExceptionWhenDecisionKeyIsNull() {
```

**Review by RomanJRW — CHANGES_REQUESTED** [review]

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            </rule>
+          </decisionTable>
+        </decision>
+      </definitions>
+      """;
+
+  @Test
+  void parseDmnModelShouldReturnModelInstanceWhenValidXml() {
+    // Given: A valid DMN XML string
+
```
[the remark this answers is not in the document]
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    // When: parseDmnModel is called
+    var result = DmnModelUtil.parseDmnModel(DMN_V1);
+
+    // Then: A valid DmnModelInstance should be returned
```
[the remark this answers is not in the document]
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  void parseDmnModelShouldThrowOptimizeRuntimeExceptionWhenInvalidXml() {
```
[the remark this answers is not in the document]
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 1 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 2 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 3 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 4 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 1 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 5 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java (hunk 6 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java]: fixed

**Review by RomanJRW — APPROVED** [review]
LGTM, just some formatting to clear up 👍

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+            </rule>
+          </decisionTable>
+        </decision>
+      </definitions>
+      """;
+
+  @Test
+  void shouldParseValidXml() {
+    // given
+
```
[the remark this answers is not in the document]
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:59]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    assertThat(result).isNotNull();
+  }
+
+  @Test
+  void shouldThrowExceptionWhenParsingInvalidXml() {
+    // given
+    final String dmnXml = "<invalid></invalid>";
+
+    // when
+
```
[the remark this answers is not in the document]
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DmnModelUtilTest.java:73]: fixed

**Pull request merged**